### PR TITLE
Fix dp optim parallel

### PR DESCRIPTION
**Pull request by wangzyphysics**

### CODE REVIEW

File: dpgen2/superop/caly_evo_step.py [hunk 1]
```diff
@@ -96,15 +97,25 @@ def __init__(
...
+            "%s--prep-dp-optim-%s-%s" % (self.inputs.parameters["block_id"], i, j)
+            for i in range(20)
+            for j in range(20)
         ]
-        self._keys = self.collect_run_calypso_keys + self.prep_run_dp_optim_keys
+        self.run_dp_optim_keys = [
+            "%s--run-dp-optim-%s-%s-%s" % (self.inputs.parameters["block_id"], i, j, k)
+            for i in range(20)
+            for j in range(20)
+            for k in range(50)
```
Comment (wanghan-iapcm) [inline dpgen2/superop/caly_evo_step.py:112]: same question

File: dpgen2/op/prep_dp_optim.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+from dpgen2.utils import (
+    BinaryFileInput,
+    set_directory,
+)
+from dpgen2.utils.run_command import (
+    run_command,
+)
+
+
+class PrepDPOptim(OP):
```
Comment (wanghan-iapcm) [inline dpgen2/op/prep_dp_optim.py:41]: It is not a general optimization op with DP. I would suggest renaming it to e.g. `PrepDPCalyOptim`

File: dpgen2/op/prep_dp_optim.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            - `caly_run_opt_file` : (`Path`)
+            - `caly_check_opt_file` : (`Path`)
+
+        """
+        group_size = ip["template_slice_config"]["group_size"]
+
+        finished = ip["finished"]
+
+        work_dir = Path(ip["task_name"])
+        # print("-----in PrepDPOptim====work_dir(task_name)", work_dir)
```
Comment (wanghan-iapcm) [inline dpgen2/op/prep_dp_optim.py:114]: plz remove all the commented prints.

File: dpgen2/op/run_dp_optim.py [hunk 4]
```diff
@@ -70,8 +62,8 @@ def get_output_sign(cls):
                 "task_name": Parameter(str),
                 "optim_results_dir": Artifact(Path),
                 "traj_results": Artifact(Path),
-                "caly_run_opt_file": Artifact(Path),
-                "caly_check_opt_file": Artifact(Path),
+                # "caly_run_opt_file": Artifact(Path),
+                # "caly_check_opt_file": Artifact(Path),
```
Comment (wanghan-iapcm) [inline dpgen2/op/run_dp_optim.py:66]: rename the comments.

**Review by zjgemi — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**